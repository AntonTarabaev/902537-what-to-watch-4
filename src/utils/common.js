import {MINUTES_IN_HOUR, MONTHS} from "@constants/main";

export const formatTime = (durationInMinutes) => {
  return (
    durationInMinutes > MINUTES_IN_HOUR ?
      `${(durationInMinutes - durationInMinutes % MINUTES_IN_HOUR) / MINUTES_IN_HOUR}h ${durationInMinutes % MINUTES_IN_HOUR}m` :
      `${durationInMinutes}m`
  );
};

export const formatDate = (date) => {
  return `${MONTHS[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`;
};
