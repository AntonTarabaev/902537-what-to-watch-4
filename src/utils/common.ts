import {MINUTES_IN_HOUR, MONTHS} from "@constants/main";
import {Film} from "@root/types";

export const formatTime = (durationInMinutes: number) => {
  return (
    durationInMinutes > MINUTES_IN_HOUR ?
      `${(durationInMinutes - durationInMinutes % MINUTES_IN_HOUR) / MINUTES_IN_HOUR}h ${durationInMinutes % MINUTES_IN_HOUR}m` :
      `${durationInMinutes}m`
  );
};

export const formatDate = (date: Date) => {
  return `${MONTHS[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`;
};

export const extend = (a: object, b: object) => {
  return Object.assign({}, a, b);
};

export const getUniqueGenres = (films: Film[], maxGenresCount: number) => {
  return [...new Set(films.map((it) => it.genre))].slice(0, maxGenresCount);
};

export const changeFormElementsDisabledProperty = (form: HTMLFormElement, disabled: boolean) => {
  Array.from(form.elements).map((formElement: HTMLInputElement) => {
    formElement.disabled = disabled;
  });
};

export const noop = () => {
  // do nothing
};
