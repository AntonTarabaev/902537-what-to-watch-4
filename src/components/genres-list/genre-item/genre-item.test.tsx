import GenreItem from "@components/genres-list/genre-item/genre-item";
import * as renderer from "react-test-renderer";
import {noop} from "@utils/common";

describe(`GenreItem component render correctly`, () => {
  it(`with active item`, () => {
    const tree = renderer.create(
        <GenreItem
          genre={`Drama`}
          activeFilter={`Drama`}
          onFilterClick={noop}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`with not active item`, () => {
    const tree = renderer.create(
        <GenreItem
          genre={`Drama`}
          activeFilter={`Thriller`}
          onFilterClick={noop}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
