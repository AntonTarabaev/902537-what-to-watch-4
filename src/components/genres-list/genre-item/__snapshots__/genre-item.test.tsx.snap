// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreItem component render correctly with active item 1`] = `
<li
  className="catalog__genres-item catalog__genres-item--active"
>
  <a
    className="catalog__genres-link"
    href={null}
    onClick={[Function]}
  >
    Drama
  </a>
</li>
`;

exports[`GenreItem component render correctly with not active item 1`] = `
<li
  className="catalog__genres-item "
>
  <a
    className="catalog__genres-link"
    href="#"
    onClick={[Function]}
  >
    Drama
  </a>
</li>
`;
